app/src/main/java/com/team/uptech/pomodoro/data/repository/impl/PomodoroRepositoryPrefsImpl.kt

```diff
@@ -0,0 +1,91 @@
+package com.team.uptech.pomodoro.data.repository.impl
+
+import android.content.Context
+import android.content.SharedPreferences
+import android.preference.PreferenceManager
+import android.util.Log
+import com.team.uptech.pomodoro.data.model.Pomodoro
+import com.team.uptech.pomodoro.data.repository.PomodoroRepository
+import io.reactivex.Completable
+import io.reactivex.Single
+import io.reactivex.schedulers.Schedulers
+import java.util.concurrent.TimeUnit
+import javax.inject.Inject
+
+/**
+ * Created on 28.04.17.
+ */
+class PomodoroRepositoryPrefsImpl @Inject constructor(context: Context) : PomodoroRepository {
+
+    private val prefs: SharedPreferences = PreferenceManager.getDefaultSharedPreferences(context)
+
+    override fun getCurrentPomodoro(): Single<Pomodoro> {
+        return Single.create<Pomodoro> { sb ->
+            val pomodoroTypeString = getPomodoroType()
+            val pomodoro = Pomodoro.valueOf(pomodoroTypeString)
+            getPomodoroTypeTime(pomodoro)
+                    .subscribe({
+                        pomodoro.time = it
+                        sb.onSuccess(pomodoro)
+                    }, {
+                        sb.onError(it)
+                        Log.e("PomodoroRepositoryPrefs", "", it)
+                    })
+        }.subscribeOn(Schedulers.io())
+                .delay(1.toLong(), TimeUnit.SECONDS) //Just for testing =)
+    }
+
+    private fun getPomodoroType() = prefs.getString(CURRENT_TYPE, TYPE_NOT_WORKING)
+
+    override fun getIsInfiniteMode(): Single<Boolean> {
+        return Single.create<Boolean> {
+            val result = prefs.getBoolean(IS_INFINITE, false)
+            it.onSuccess(result)
+        }.subscribeOn(Schedulers.io())
+    }
+
+    override fun getPomodoroTypeTime(pomodoro: Pomodoro): Single<Int> {
+        return Single.create<Int> {
+            when (pomodoro) {
+                Pomodoro.WORK -> it.onSuccess(prefs.getInt(pomodoro.name, DEFAULT_WORK_TIME))
+                Pomodoro.BREAK -> it.onSuccess(prefs.getInt(pomodoro.name, DEFAULT_BREAK_TIME))
+                Pomodoro.NOT_WORKING -> it.onSuccess(prefs.getInt(pomodoro.name, 0))
```

Comment:
you could move DEFAULT_WORK_TIME, DEFAULT_BREAK_TIME, 0 to the Pomodoro enum and this "when" will turn into a one-line statement